### DISCUSSION TITLE
[ISSUE]: The count of (GitVersion.PreReleaseNumber) is reset to 0 in the branch Develop without any reason.

### DISCUSSION THREAD

**Discussion by kornitdevops** [GitTools/GitVersion#4101] [Q&A]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

AzureDevops task

### GitVersion version

5.x

### Operating system

Windows

### What are you seeing?

The count of (GitVersion.PreReleaseNumber) is reset to 0 in the branch Develop without any reason.

### What is expected?

it will continue to count above 4.5.0.47, instead its downgraded to 4.5.0.2

### Steps to Reproduce

just merge "personal/branch" to the Develop branch with 1 commit
![Screenshot 2024-07-16 164345](https://github.com/user-attachments/assets/dfb56e42-a3d7-4060-b9d8-85182e46abaf)

### Output log or link to your CI build (if appropriate).

```shell
gitvesion.yml:

        mode: ContinuousDeployment
        tag-prefix: '[vV]'
        legacy-semver-padding: 5
        continuous-delivery-fallback-tag: '' # ci
        major-version-bump-message: '\+semver:\s?(breaking|major)'
        minor-version-bump-message: '\+semver:\s?(feature|minor)'
        patch-version-bump-message: '\+semver:\s?(fix|patch)'
        no-bump-message: '\+semver:\s?(none|skip)'
        commit-message-incrementing: Enabled
        branches:
          main:
            regex: ^main$
            tag: ''
            increment: None
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
            - release
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: true
          develop:
            regex: ^develop$
            tag: dev
            increment: Minor
            prevent-increment-of-merged-branch-version: true
            tracks-release-branches: true
            track-merge-target: true
          release:
            regex: ^release?[/-]\d.\d.\d$
            tag: release
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
            is-release-branch: true
          feature:
            regex: ^feature[/-]
            tag: f-{BranchName}
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
          pull-request:
            tag: '' # PullRequest
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            tag-number-pattern: '[/-](?<number>\d+)'
            regex: ^(pull|pull\-requests|pr)[/-]
            source-branches:
            - develop
            - main
            - release
            - feature
            - hotfix
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: false
          hotfix:
            regex: ^hotfix[/-]
            tag: h-{BranchName}
            increment: Patch
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - main
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: false
        ignore:
          sha: []
        commit-date-format: dd-MM-yyyy
        merge-message-formats: {}
        update-build-number: true


pipeline 4.5.00.47:
 47 commits found between c3aeaad Increment Version ***NO_CI*** and f2d3294 Merged PR 23489: 
pipeline 4.5.0.2:
```

**Comment by arturcic** [comment]
Currently the 5.x versions is really old and we are not maintaining it. We're in the process of releasing a new v6.0.0 version so you better target the new version. Please check this example https://github.com/GitTools/actions/blob/main/docs/examples/azure/gitversion/setup/usage-examples.md#example-3

**Reply by kornitdevops** [reply]
Thank you for the answer 

I updated the gitversion to 6X and now I get an error:
do I need to change something in the Gitversion.yml because of version 6?

**updated template:**

  - task: PowerShell@2
    inputs:
      targetType: 'inline'
      script: |
        $directoryPath = "$(repositoryPath)/.azuredevops/gitversion"
        $filePath = "$directoryPath/GitVersion.yml"
        
        # Check if the file exists
        if (Test-Path -Path $filePath) {
            Write-Output "GitVersion.yml already exists. No action needed."
        } else {
            Write-Output "GitVersion.yml does not exist. Creating directory and file."
        
            # If the file does not exist, ensure the directory exists
            if (-not (Test-Path -Path $directoryPath)) {
                New-Item -ItemType Directory -Path $directoryPath -Force
            }
        
            # Define the content of the GitVersion.yml file
            $content = @"
        mode: ContinuousDeployment
        tag-prefix: '[vV]'
        legacy-semver-padding: 5
        continuous-delivery-fallback-tag: '' # ci
        major-version-bump-message: '\+semver:\s?(breaking|major)'
        minor-version-bump-message: '\+semver:\s?(feature|minor)'
        patch-version-bump-message: '\+semver:\s?(fix|patch)'
        no-bump-message: '\+semver:\s?(none|skip)'
        commit-message-incrementing: Enabled
        branches:
          main:
            regex: ^main$
            tag: ''
            increment: None
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
            - release
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: true
          develop:
            regex: ^develop$
            tag: dev
            increment: Minor
            prevent-increment-of-merged-branch-version: true
            tracks-release-branches: true
            track-merge-target: true
          release:
            regex: ^release?[/-]\d.\d.\d$
            tag: release
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
            is-release-branch: true
          feature:
            regex: ^feature[/-]
            tag: f-{BranchName}
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - develop
          pull-request:
            tag: '' # PullRequest
            increment: Inherit
            prevent-increment-of-merged-branch-version: true
            tag-number-pattern: '[/-](?<number>\d+)'
            regex: ^(pull|pull\-requests|pr)[/-]
            source-branches:
            - develop
            - main
            - release
            - feature
            - hotfix
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: false
          hotfix:
            regex: ^hotfix[/-]
            tag: h-{BranchName}
            increment: Patch
            prevent-increment-of-merged-branch-version: true
            source-branches:
            - main
            tracks-release-branches: false
            is-release-branch: false
            is-mainline: false
        ignore:
          sha: []
        commit-date-format: dd-MM-yyyy
        merge-message-formats: {}
        update-build-number: true
        "@
            # Write the content to the file
            $content | Out-File -FilePath $filePath -Encoding utf8
        }
    displayName: Create GitVersion.yml

  - task: gitversion/setup@0
    inputs:
      versionSpec: '6.x'
      includePrerelease: true
      
  - task: gitversion/execute@0
    inputs:
      targetPath: '$(repositoryPath)'
      useConfigFile: true
      configFilePath: '$(repositoryPath)/.azuredevops/gitversion/GitVersion.yml'
      
      
  **log:** 
  
INFO [24-07-17 9:40:44:35] Found configuration file at 'D:\Agents\A1\277\s\.azuredevops\gitversion\GitVersion.yml'
WARN [24-07-17 9:40:44:51] An error occurred:
Could not build the configuration instance because following exception occurred: 'Property 'tag' not found on type 'GitVersion.Configuration.BranchConfiguration'.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.

**Reply by kornitdevops** [reply]
I have updated the Yml:

        mode: ContinuousDelivery
        tag-prefix: '[vV]'
        major-version-bump-message: '\+semver:\s?(breaking|major)'
        minor-version-bump-message: '\+semver:\s?(feature|minor)'
        patch-version-bump-message: '\+semver:\s?(fix|patch)'
        no-bump-message: '\+semver:\s?(none|skip)'
        commit-message-incrementing: Enabled
        branches:
          main:
            regex: ^main$
            label: ''
            increment: None
            source-branches:
            - develop
            - release
            tracks-release-branches: false
            is-release-branch: false
            is-main-branch: true
          develop:
            regex: ^develop$
            label: dev
            increment: Minor
            tracks-release-branches: true
            track-merge-target: true
          release:
            regex: ^release?[/-]\d.\d.\d$
            label: release
            increment: Inherit
            source-branches:
            - develop
            is-release-branch: true
          feature:
            regex: ^features?[/-](?<BranchName>.+)
            label: f-{BranchName}
            increment: Inherit
            source-branches:
            - develop
          pull-request:
            label: '' # PullRequest
            increment: Inherit
            label-number-pattern: '[/-](?<number>\d+)'
            regex: ^(pull|pull\-requests|pr)[/-]
            source-branches:
            - develop
            - main
            - release
            - feature
            - hotfix
            tracks-release-branches: false
            is-release-branch: false
            is-main-branch: false
          hotfix:
            regex: ^hotfix[/-]
            label: h-{BranchName}
            increment: Patch
            source-branches:
            - main
            tracks-release-branches: false
            is-release-branch: false
            is-main-branch: false
        ignore:
          sha: []
        commit-date-format: dd-MM-yyyy
        merge-message-formats: {}
        update-build-number: true
        "@
            # Write the content to the file
            $content | Out-File -FilePath $filePath -Encoding utf8
        }
        
        
and now it works. 

but the version is 0.0.0:


Executing GenerateBuildLogOutput for 'AzurePipelines'.
{
  "AssemblySemFileVer": "0.1.0.0",
  "AssemblySemVer": "0.1.0.0",
  "BranchName": "feature/c",
  "BuildMetaData": 723,
  "CommitDate": "17-07-2024",
  "CommitsSinceVersionSource": 723,
  "EscapedBranchName": "feature-c",
  "FullBuildMetaData": "723.Branch.feature-c.Sha.d623e273f2b2cfc8f5ed0e32f9b76f42131aed8e",
  "FullSemVer": "0.1.0-f-c.1+723",
  "InformationalVersion": "0.1.0-f-c.1+723.Branch.feature-c.Sha.d623e273f2b2cfc8f5ed0e32f9b76f42131aed8e",
  "Major": 0,
  "MajorMinorPatch": "0.1.0",
  "Minor": 1,
  "Patch": 0,
  "PreReleaseLabel": "f-c",
  "PreReleaseLabelWithDash": "-f-c",
  "PreReleaseNumber": 1,
  "PreReleaseTag": "f-c.1",
  "PreReleaseTagWithDash": "-f-c.1",
  "SemVer": "0.1.0-f-c.1",
  "Sha": "d623e273f2b2cfc8f5ed0e32f9b76f42131aed8e",
  "ShortSha": "d623e27",
  "UncommittedChanges": 1,
  "VersionSourceSha": "",
  "WeightedPreReleaseNumber": 30001
}

**Comment by HHobeck** [comment]
I'm going to close this discussion because gitversion version 5.x is not supported anymore. Please try the latest version.

Happy Branching!
